File: DFS, BFS/1260.py
#DFS와 BFS
from collections import deque
def dfs(graph, start, visited):
    visited[start] = True
    print(start, end=" ")
    for i in graph[start]:
        if not visited[i]:
            dfs(graph, i, visited)

def bfs(graph, start, visited):
    queue = deque([start])
    visited[start] = True
    while queue:
        v = queue.popleft()
        print(v, end=" ")
        for i in graph[v]:
            if not visited[i]:
                queue.append(i)
                visited[i] = True

if __name__ == "__main__":
    n, m, v = map(int, input().split())
    graph = [[] for _ in range(n+1)]
    visited = [False]*(n+1)
    for _ in range(m):
        a, b = map(int, input().split())
        graph[a].append(b)
        graph[b].append(a)
    for i in range(1, n+1):
        graph[i].sort()
    dfs(graph, v, visited)
    visited = [False]*(n+1)
    print()
    bfs(graph, v, visited)

File: DFS, BFS/1260.js
// DFS와 BFS
let input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

let [n, m, root] = input[0].split(" ").map((v) => Number(v));
let graph = Array.from(Array(n + 1), () => Array());
let visited = Array(n + 1).fill(0);
let result = ``;

for (let i = 1; i < input.length; i++) {
  let temp = input[i].split(" ").map((v) => Number(v));
  graph[temp[0]].push(temp[1]);
  graph[temp[1]].push(temp[0]);
}
graph.map((v) => v.sort((a, b) => a - b));
dfs(root);
result += "\n";
visited = visited.map((v) => 0);
bfs(root);
console.log(result);

function dfs(v) {
  result += `${v} `;
  visited[v] = 1;
  for (let nv of graph[v]) {
    if (!visited[nv]) {
      dfs(nv);
    }
  }
}

function bfs(v) {
  let q = [];
  visited[v] = 1;
  q.push(v);
  result += `${v} `;
  while (q.length) {
    let n = q.shift();
    for (let nv of graph[n]) {
      if (!visited[nv]) {
        visited[nv] = 1;
        q.push(nv);
        result += `${nv} `;
      }
    }
  }
}
